src/service/mercury/index.ts

```diff
@@ -564,48 +566,34 @@ export class MercuryClient {
       throw new Error(ERROR.UNSUPPORTED_NETWORK);
     }
 
-    let balances: any = null;
-    let isFunded = null;
-    let subentryCount = 0;
-
-    try {
-      const server = new Horizon.Server(networkUrl, {
-        allowHttp: !networkUrl.includes("https"),
-      });
-      const resp = await fetchAccountDetails(pubKey, server);
-      balances = resp.balances;
-      subentryCount = resp.subentryCount;
-
-      for (let i = 0; i < Object.keys(resp.balances).length; i++) {
-        const k = Object.keys(resp.balances)[i];
-        const v: any = resp.balances[k];
-        if (v.liquidity_pool_id) {
-          const lp = await server
-            .liquidityPools()
-            .liquidityPoolId(v.liquidity_pool_id)
-            .call();
-          balances[k] = {
-            ...balances[k],
-            liquidityPoolId: v.liquidity_pool_id,
-            reserves: lp.reserves,
-          };
-          delete balances[k].liquidity_pool_id;
-        }
+    const server = new Horizon.Server(networkUrl, {
+      allowHttp: !networkUrl.includes("https"),
+    });
+    const resp = await fetchAccountDetails(pubKey, server);
+
+    for (let i = 0; i < Object.keys(resp.balances).length; i++) {
+      const k = Object.keys(resp.balances)[i];
+      const v = resp.balances[k];
+      if ("liquidity_pool_id" in v) {
+        const _v = v as any as Horizon.HorizonApi.BalanceLineLiquidityPool;
```

Comment:
This code has lived for a long time and it looks like there are paths in the client that expect this but the types seem to disagree that there would ever be a `liquidity_pool_id` in a `Balance(AssetBalance | NativeBalance)`.